Integration tests failing on macOS

**Describe the bug**
When running `make docker-test-integration` on macOS, I get many errors:
```
==> Running integration tests <==
SOME TESTS WILL FAIL IF NOT RUN AS ROOT!
go test github.com/vmware-tanzu/antrea/test/integration/...
?   	github.com/vmware-tanzu/antrea/test/integration	[no test files]
I0603 01:41:24.811232     204 server.go:616] Reconciliation for CNI server
I0603 01:41:24.812979     204 server.go:351] Received CmdAdd request cni_args:<container_id:"dummy-0" netns:"/var/run/netns/cnitest-093f1ed0-b5b1-5be8-1484-14ca7cf40f34" ifname:"eth0" args:"IgnoreUnknown=1;K8S_POD_NAMESPACE=t1;K8S_POD_NAME=test-1;K8S_POD_INFRA_CONTAINER_ID=test-111111" network_configuration:"{\n\t\"cniVersion\": \"0.4.0\",\n\t\"name\": \"testConfig\",\n\t\"type\": \"antrea\",\n\t\"isDefaultGateway\": true,\n    \"ipam\": {\n        \"type\":    \"mock\",\n        \"dataDir\": \"/tmp/antrea_server_test254160683\",\n        \"ranges\": [\n            [{\n                \"subnet\":  \"10.1.2.0/24\"\n            }]\n        ]\n    }\n}" >
I0603 01:41:24.813368     204 server.go:191] Load network configurations: container_id:"dummy-0" netns:"/var/run/netns/cnitest-093f1ed0-b5b1-5be8-1484-14ca7cf40f34" ifname:"eth0" args:"IgnoreUnknown=1;K8S_POD_NAMESPACE=t1;K8S_POD_NAME=test-1;K8S_POD_INFRA_CONTAINER_ID=test-111111" network_configuration:"{\"cniVersion\":\"0.4.0\",\"name\":\"testConfig\",\"type\":\"antrea\",\"dns\":{},\"ipam\":{\"type\":\"mock\",\"subnet\":\"192.168.1.0/24\",\"gateway\":\"192.168.1.1\"},\"runtimeConfig\":{\"dns\":{}}}"
I0603 01:41:24.813422     204 server.go:405] Added ip addresses from IPAM driver, IP:[{Version:4 Interface:<nil> Address:{IP:10.1.2.100 Mask:ffffff00} Gateway:10.1.2.1}], Routes:[{Dst:{IP:10.0.0.0 Mask:ff000000} GW:10.1.2.1} {Dst:{IP:0.0.0.0 Mask:00000000} GW:10.1.2.1}], DNS:{Nameservers:[] Domain: Search:[] Options:[]}
I0603 01:41:24.824342     204 pod_configuration.go:244] Configured interfaces for container dummy-0
I0603 01:41:24.824522     204 server.go:433] CmdAdd succeeded
--- FAIL: TestAntreaServerFunc (0.03s)
    --- FAIL: TestAntreaServerFunc/ADD/DEL/CHECK_for_0.4.0_config (0.03s)
        cniserver_test.go:420:
            	Error Trace:	cniserver_test.go:420
            	            				cniserver_test.go:570
            	            				cniserver_test.go:655
            	Error:      	"[%!s(*netlink.Device=&{{1 65536 1000 lo [] 4 8 0 0 <nil>  0xc000438000 0 0xc0004161a0 loopback <nil> 2 0 1 1 65536 65535 [] 0 <nil>}}) %!s(*netlink.Iptun=&{{2 1480 1000 tunl0 [] 0 128 0 0 <nil>  0xc000438180 0 0xc0004161c0 ipip <nil> 2 0 1 1 65536 65535 [] 0 <nil>} 0 0 0 0 [0 0 0 0] [0 0 0 0] 0 0 0 0 false}) %!s(*netlink.Ip6tnl=&{{3 1452 1000 ip6tnl0 [] 0 128 0 0 <nil>  0xc000438300 0 0xc0004161e0 tunnel6 <nil> 2 0 1 1 65536 65535 [] 0 <nil>} 0 [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0] [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0] 0 0 0 262144 41 0}) %!s(*netlink.Veth=&{{5 1450 0 eth0 [190 43 164 25 224 74] 19 69699 4 0 <nil>  0xc000438480 0 0xc000416200 ether <nil> 6 0 1 1 65536 65535 [] 0 <nil>}  []})]" should have 2 item(s), but has 4
            	Test:       	TestAntreaServerFunc/ADD/DEL/CHECK_for_0.4.0_config
I0603 01:41:24.855453     204 server.go:616] Reconciliation for CNI server
I0603 01:41:24.855838     204 server.go:351] Received CmdAdd request cni_args:<container_id:"dummy-0" netns:"/var/run/netns/cnitest-28276bc1-61e6-3096-1add-021c2f1cb0f2" ifname:"eth0" args:"IgnoreUnknown=1;K8S_POD_NAMESPACE=t1;K8S_POD_NAME=test-1;K8S_POD_INFRA_CONTAINER_ID=test-111111" network_configuration:"{\n\"cniVersion\":\"0.4.0\",\n\"name\":\"azure\",\n\"prevResult\":\n  {\n    \"cniVersion\":\"0.4.0\",\n    \"interfaces\":\n     [\n       {\"name\":\"eth0\"},\n       {\"name\":\"eth0\"}\n     ],\n    \"ips\":\n     [\n       {\n         \"version\":\"4\",\n         \"address\":\"10.10.10.2/24\",\n         \"gateway\":\"10.10.10.1\"}\n     ],\n    \"routes\":\n     [\n       {\n         \"dst\":\"10.0.0.0/8\",\n         \"gw\":\"10.10.10.1\"\n       }\n     ],\n     \"dns\":\n     {\n       \"nameservers\":\n        [\n          \"[8.8.8.8]\"\n        ]\n     }\n  } \n}" >
I0603 01:41:24.855934     204 server.go:191] Load network configurations: container_id:"dummy-0" netns:"/var/run/netns/cnitest-28276bc1-61e6-3096-1add-021c2f1cb0f2" ifname:"eth0" args:"IgnoreUnknown=1;K8S_POD_NAMESPACE=t1;K8S_POD_NAME=test-1;K8S_POD_INFRA_CONTAINER_ID=test-111111" network_configuration:"{\n\"cniVersion\":\"0.4.0\",\n\"name\":\"azure\",\n\"prevResult\":\n  {\n    \"cniVersion\":\"0.4.0\",\n    \"interfaces\":\n     [\n       {\"name\":\"eth0\"},\n       {\"name\":\"eth0\"}\n     ],\n    \"ips\":\n     [\n       {\n         \"version\":\"4\",\n         \"address\":\"10.10.10.2/24\",\n         \"gateway\":\"10.10.10.1\"}\n     ],\n    \"routes\":\n     [\n       {\n         \"dst\":\"10.0.0.0/8\",\n         \"gw\":\"10.10.10.1\"\n       }\n     ],\n     \"dns\":\n     {\n       \"nameservers\":\n        [\n          \"[8.8.8.8]\"\n        ]\n     }\n  } \n}"
I0603 01:41:24.856025     204 server.go:574] CNI Chaining: add
I0603 01:41:24.856812     204 server.go:441] Received CmdDel request cni_args:<container_id:"dummy-0" netns:"/var/run/netns/cnitest-28276bc1-61e6-3096-1add-021c2f1cb0f2" ifname:"eth0" args:"IgnoreUnknown=1;K8S_POD_NAMESPACE=t1;K8S_POD_NAME=test-1;K8S_POD_INFRA_CONTAINER_ID=test-111111" network_configuration:"{\n\"cniVersion\":\"0.4.0\",\n\"name\":\"azure\",\n\"prevResult\":\n  {\n    \"cniVersion\":\"0.4.0\",\n    \"interfaces\":\n     [\n       {\"name\":\"eth0\"},\n       {\"name\":\"eth0\"}\n     ],\n    \"ips\":\n     [\n       {\n         \"version\":\"4\",\n         \"address\":\"10.10.10.2/24\",\n         \"gateway\":\"10.10.10.1\"}\n     ],\n    \"routes\":\n     [\n       {\n         \"dst\":\"10.0.0.0/8\",\n         \"gw\":\"10.10.10.1\"\n       }\n     ],\n     \"dns\":\n     {\n       \"nameservers\":\n        [\n          \"[8.8.8.8]\"\n        ]\n     }\n  } \n}" >
I0603 01:41:24.856898     204 server.go:191] Load network configurations: container_id:"dummy-0" netns:"/var/run/netns/cnitest-28276bc1-61e6-3096-1add-021c2f1cb0f2" ifname:"eth0" args:"IgnoreUnknown=1;K8S_POD_NAMESPACE=t1;K8S_POD_NAME=test-1;K8S_POD_INFRA_CONTAINER_ID=test-111111" network_configuration:"{\n\"cniVersion\":\"0.4.0\",\n\"name\":\"azure\",\n\"prevResult\":\n  {\n    \"cniVersion\":\"0.4.0\",\n    \"interfaces\":\n     [\n       {\"name\":\"eth0\"},\n       {\"name\":\"eth0\"}\n     ],\n    \"ips\":\n     [\n       {\n         \"version\":\"4\",\n         \"address\":\"10.10.10.2/24\",\n         \"gateway\":\"10.10.10.1\"}\n     ],\n    \"routes\":\n     [\n       {\n         \"dst\":\"10.0.0.0/8\",\n         \"gw\":\"10.10.10.1\"\n       }\n     ],\n     \"dns\":\n     {\n       \"nameservers\":\n        [\n          \"[8.8.8.8]\"\n        ]\n     }\n  } \n}"
I0603 01:41:24.856932     204 server.go:599] CNI Chaining: delete
I0603 01:41:24.856965     204 pod_configuration.go:514] Removed interfaces for container dummy-0
I0603 01:41:24.884780     204 server.go:351] Received CmdAdd request cni_args:<container_id:"dummy-0" netns:"/var/run/netns/cnitest-ee1a3571-db68-01b4-6b2f-46191d1269e6" ifname:"eth0" args:"IgnoreUnknown=1;K8S_POD_NAMESPACE=t1;K8S_POD_NAME=test-1;K8S_POD_INFRA_CONTAINER_ID=test-111111" network_configuration:"{\n\"cniVersion\":\"0.4.0\",\n\"name\":\"azure\",\n\"prevResult\":\n  {\n    \"cniVersion\":\"0.4.0\",\n    \"interfaces\":\n     [\n       {\"name\":\"eth0\"},\n       {\"name\":\"eth0\"}\n     ],\n    \"ips\":\n     [\n       {\n         \"version\":\"4\",\n         \"address\":\"10.10.10.2/24\",\n         \"gateway\":\"10.10.10.1\"}\n     ],\n    \"routes\":\n     [\n       {\n         \"dst\":\"10.0.0.0/8\",\n         \"gw\":\"10.10.10.1\"\n       }\n     ],\n     \"dns\":\n     {\n       \"nameservers\":\n        [\n          \"[8.8.8.8]\"\n        ]\n     }\n  } \n}" >
I0603 01:41:24.886710     204 server.go:191] Load network configurations: container_id:"dummy-0" netns:"/var/run/netns/cnitest-ee1a3571-db68-01b4-6b2f-46191d1269e6" ifname:"eth0" args:"IgnoreUnknown=1;K8S_POD_NAMESPACE=t1;K8S_POD_NAME=test-1;K8S_POD_INFRA_CONTAINER_ID=test-111111" network_configuration:"{\n\"cniVersion\":\"0.4.0\",\n\"name\":\"azure\",\n\"prevResult\":\n  {\n    \"cniVersion\":\"0.4.0\",\n    \"interfaces\":\n     [\n       {\"name\":\"eth0\"},\n       {\"name\":\"eth0\"}\n     ],\n    \"ips\":\n     [\n       {\n         \"version\":\"4\",\n         \"address\":\"10.10.10.2/24\",\n         \"gateway\":\"10.10.10.1\"}\n     ],\n    \"routes\":\n     [\n       {\n         \"dst\":\"10.0.0.0/8\",\n         \"gw\":\"10.10.10.1\"\n       }\n     ],\n     \"dns\":\n     {\n       \"nameservers\":\n        [\n          \"[8.8.8.8]\"\n        ]\n     }\n  } \n}"
I0603 01:41:24.886769     204 server.go:574] CNI Chaining: add
I0603 01:41:24.888399     204 server.go:441] Received CmdDel request cni_args:<container_id:"dummy-0" netns:"/var/run/netns/cnitest-ee1a3571-db68-01b4-6b2f-46191d1269e6" ifname:"eth0" args:"IgnoreUnknown=1;K8S_POD_NAMESPACE=t1;K8S_POD_NAME=test-1;K8S_POD_INFRA_CONTAINER_ID=test-111111" network_configuration:"{\n\"cniVersion\":\"0.4.0\",\n\"name\":\"azure\",\n\"prevResult\":\n  {\n    \"cniVersion\":\"0.4.0\",\n    \"interfaces\":\n     [\n       {\"name\":\"eth0\"},\n       {\"name\":\"eth0\"}\n     ],\n    \"ips\":\n     [\n       {\n         \"version\":\"4\",\n         \"address\":\"10.10.10.2/24\",\n         \"gateway\":\"10.10.10.1\"}\n     ],\n    \"routes\":\n     [\n       {\n         \"dst\":\"10.0.0.0/8\",\n         \"gw\":\"10.10.10.1\"\n       }\n     ],\n     \"dns\":\n     {\n       \"nameservers\":\n        [\n          \"[8.8.8.8]\"\n        ]\n     }\n  } \n}" >
I0603 01:41:24.888475     204 server.go:191] Load network configurations: container_id:"dummy-0" netns:"/var/run/netns/cnitest-ee1a3571-db68-01b4-6b2f-46191d1269e6" ifname:"eth0" args:"IgnoreUnknown=1;K8S_POD_NAMESPACE=t1;K8S_POD_NAME=test-1;K8S_POD_INFRA_CONTAINER_ID=test-111111" network_configuration:"{\n\"cniVersion\":\"0.4.0\",\n\"name\":\"azure\",\n\"prevResult\":\n  {\n    \"cniVersion\":\"0.4.0\",\n    \"interfaces\":\n     [\n       {\"name\":\"eth0\"},\n       {\"name\":\"eth0\"}\n     ],\n    \"ips\":\n     [\n       {\n         \"version\":\"4\",\n         \"address\":\"10.10.10.2/24\",\n         \"gateway\":\"10.10.10.1\"}\n     ],\n    \"routes\":\n     [\n       {\n         \"dst\":\"10.0.0.0/8\",\n         \"gw\":\"10.10.10.1\"\n       }\n     ],\n     \"dns\":\n     {\n       \"nameservers\":\n        [\n          \"[8.8.8.8]\"\n        ]\n     }\n  } \n}"
I0603 01:41:24.888505     204 server.go:599] CNI Chaining: delete
I0603 01:41:24.888558     204 pod_configuration.go:514] Removed interfaces for container dummy-0
I0603 01:41:24.908752     204 ofctrl.go:170] Initialize connection or re-connect to /var/run/openvswitch/br01.mgmt.
E0603 01:41:25.910362     204 ofctrl.go:216] Failed to connect to /var/run/openvswitch/br01.mgmt, retry after 1s: dial unix /var/run/openvswitch/br01.mgmt: connect: no such file or directory.
E0603 01:41:26.911271     204 ofctrl.go:216] Failed to connect to /var/run/openvswitch/br01.mgmt, retry after 1s: dial unix /var/run/openvswitch/br01.mgmt: connect: no such file or directory.
E0603 01:41:27.913916     204 ofctrl.go:216] Failed to connect to /var/run/openvswitch/br01.mgmt, retry after 1s: dial unix /var/run/openvswitch/br01.mgmt: connect: no such file or directory.
E0603 01:41:28.914772     204 ofctrl.go:216] Failed to connect to /var/run/openvswitch/br01.mgmt, retry after 1s: dial unix /var/run/openvswitch/br01.mgmt: connect: no such file or directory.
I0603 01:41:29.910688     204 ofctrl.go:218] Controller is deleted, stop re-connections
--- FAIL: TestConnectivityFlows (5.09s)
    openflow_test.go:215: Failed to initialize openflow client: failed to connect to OpenFlow switch after 5 seconds
    openflow_test_utils.go:56: Failed to install flow:
        &{priority=0 drop}
        Existing flows:
        []
...
```

**To Reproduce**
run `make docker-test-integration` on macOS after cloning the latest Antrea.

**Expected**
Since `make docker-test-integration` runs the integration tests in a Docker image, I expected the integrations tests to run successfully. My guess is that the tests make some assumptions (with respect to the network configuration) which do not hold in the HyperKit VM used by Docker Desktop on macOS.

**Actual behavior**
Test failures.

**Versions:**
Antrea v0.8.0-dev

--- a/test/integration/agent/cniserver_test.go
+++ b/test/integration/agent/cniserver_test.go
@@ -417,7 +417,21 @@ func (tester *cmdAddDelTester) cmdAddTest(tc testCase, dataDir string) (*current
 		return err
 	})
 	testRequire.Nil(err)
-	testRequire.Len(linkList, 2)
+	// when running the integration tests in a Docker image on macOS (Docker
+	// Desktop), we observe that the namespace includes 4 links, not 2
+	// (localhost and veth). It seems that the HyperKit Linux VM loads the
+	// ipip module, which means that the Linux Kernel creates 2 extra
+	// interfaces in each namespace (tunl0 and ip6tnl).
+	testRequire.GreaterOrEqual(len(linkList), 2)
+	// check that the list includes the eth0 veth
+	vethFound := false
+	for _, link := range linkList {
+		if link.Type() == "veth" && link.Attrs().Name == IFName {
+			vethFound = true
+			break
+		}
+	}
+	testRequire.Truef(vethFound, "Missing '%s' veth in target namespace", IFName)
 
 	// Find the veth peer in the container namespace and the default route.
 	tester.checkContainerNetworking(tc)

--- a/test/integration/ovs/openflow_test_utils.go
+++ b/test/integration/ovs/openflow_test_utils.go
@@ -25,7 +25,10 @@ import (
 )
 
 func PrepareOVSBridge(brName string) error {
-	cmdStr := fmt.Sprintf("ovs-vsctl --may-exist add-br %s -- set Bridge %s protocols='OpenFlow10,OpenFlow13'", brName, brName)
+	// using the netdev datapath type does not impact test coverage but
+	// ensures that the integration tests can be run with Docker Desktop on
+	// macOS.
+	cmdStr := fmt.Sprintf("ovs-vsctl --may-exist add-br %s -- set Bridge %s protocols='OpenFlow10,OpenFlow13' datapath_type=netdev", brName, brName)
 	err := exec.Command("/bin/sh", "-c", cmdStr).Run()
 	if err != nil {
 		return err

--- a/test/integration/ovs/ovs_client_test.go
+++ b/test/integration/ovs/ovs_client_test.go
@@ -59,7 +59,10 @@ func (data *testData) setup(t *testing.T) {
 		t.Fatalf("Could not establish connection to %s after %s", UDSAddress, defaultConnectTimeout)
 	}
 
-	data.br = ovsconfig.NewOVSBridge(bridgeName, "system", data.ovsdb)
+	// using the netdev datapath type does not impact test coverage but
+	// ensures that the integration tests can be run with Docker Desktop on
+	// macOS.
+	data.br = ovsconfig.NewOVSBridge(bridgeName, "netdev", data.ovsdb)
 	err = data.br.Create()
 	require.Nil(t, err, "Failed to create bridge %s", bridgeName)
 }

